server/src/main/java/org/cloudfoundry/identity/uaa/scim/ScimUserAliasHandler.java

```diff
@@ -0,0 +1,158 @@
+        if (idpInCurrentZone.isEmpty()) {
+            return false;
+        }
+        return EntityAliasHandler.isValidAliasPair(idpInCurrentZone.get(), idpInAliasZone.get());
+    }
+
+    @Override
+    protected void setPropertiesFromExistingAliasEntity(
+            final ScimUser newAliasEntity,
+            final ScimUser existingAliasEntity
+    ) {
+        // these three timestamps should not be overwritten by the timestamps of the original user
+        newAliasEntity.setPasswordLastModified(existingAliasEntity.getPasswordLastModified());
+        newAliasEntity.setLastLogonTime(existingAliasEntity.getLastLogonTime());
+        newAliasEntity.setPreviousLogonTime(existingAliasEntity.getPreviousLogonTime());
+    }
+
+    private Optional<IdentityProvider<?>> retrieveIdpByOrigin(final String originKey, final String zoneId) {
+        final IdentityProvider<?> idpInAliasZone;
+        try {
+            idpInAliasZone = identityProviderProvisioning.retrieveByOrigin(originKey, zoneId);
+        } catch (final EmptyResultDataAccessException e) {
+            return Optional.empty();
+        }
+        return Optional.ofNullable(idpInAliasZone);
+    }
+
+    @Override
+    protected void setId(final ScimUser entity, final String id) {
+        entity.setId(id);
+    }
+
+    @Override
+    protected void setZoneId(final ScimUser entity, final String zoneId) {
+        entity.setZoneId(zoneId);
+    }
+
+    @Override
+    protected ScimUser cloneEntity(final ScimUser originalEntity) {
+        final ScimUser aliasUser = new ScimUser();
+
+        aliasUser.setId(null);
+        aliasUser.setExternalId(originalEntity.getExternalId());
+
+        /* we only allow alias users to be created if their origin IdP has an alias to the same zone, therefore, an IdP
+         * with the same origin key will exist in the alias zone  */
+        aliasUser.setOrigin(originalEntity.getOrigin());
+
+        aliasUser.setUserName(originalEntity.getUserName());
+        aliasUser.setName(new ScimUser.Name(originalEntity.getGivenName(), originalEntity.getFamilyName()));
+
+        aliasUser.setEmails(originalEntity.getEmails());
+        aliasUser.setPhoneNumbers(originalEntity.getPhoneNumbers());
+
+        aliasUser.setActive(originalEntity.isActive());
+        aliasUser.setVerified(originalEntity.isVerified());
+
+        // idzId and alias properties will be set later
+        aliasUser.setZoneId(null);
+        aliasUser.setAliasId(null);
+        aliasUser.setAliasZid(null);
+
+        /* these timestamps will be overwritten:
+         *  - creation: with current timestamp during persistence (JdbcScimUserProvisioning)
+         *  - update: with values from existing alias entity */
+        aliasUser.setPasswordLastModified(null);
+        aliasUser.setLastLogonTime(null);
+        aliasUser.setPreviousLogonTime(null);
```

Comment:
We do in other places Json object copy, e.g.

ScimUser scimCopy = JsonUtils.convertValue(originalEntity, ScimUser.class);

Reply:
see the comment above